client/me/billing-history/receipt.jsx

```diff
@@ -142,7 +143,13 @@ class BillingReceipt extends React.Component {
 
 	renderLineItems() {
 		const { transaction, translate } = this.props;
-		const items = transaction.items.map( item => {
+
+		const groupedTransactionItems = [
+			...reject( transaction.items, { product: 'Domain' } ),
```

Comment:
Maybe we could create a helper for this to avoid repeating it bellow. Also adding a comment explaining why are we removing the domain products might be helpful, because it might not be obvious to someone reading this without proper context.